source/extensions/transport_sockets/tcp_stats/tcp_stats.cc

```diff
@@ -78,8 +78,14 @@ absl::optional<struct tcp_info> TcpStatsSocket::querySocketInfo() {
   struct tcp_info info;
   memset(&info, 0, sizeof(info));
   socklen_t optlen = sizeof(info);
+
+  // Minimal supported size of the `struct tcp_info`. We rely on that field at build time, so it
+  // must be present in the kernel at runtime.
+  size_t size = offsetof(struct tcp_info, LAST_TCP_INFO_FIELD_WE_USE) +
```

Comment:
I think it would be even better if you set `optlen = size;`, because then the kernel won't fill in later fields, so if someone accidentally adds a use of one of those fields it will always be zero, making it obvious that there's a bug.

Reply:
agreed